plugin.rb

```diff
@@ -343,4 +343,31 @@ class Engine < ::Rails::Engine
 
   register_notification_consolidation_plan(reacted_by_two_users)
   register_notification_consolidation_plan(consolidated_reactions)
+
+  on(:first_post_moved) do |target_post, original_post|
+    reaction_cols = (DiscourseReactions::Reaction.column_names - %w[id post_id]).join(", ")
+    reaction_user_cols =
+      (DiscourseReactions::ReactionUser.column_names - %w[id post_id reaction_id]).join(", ")
+
+    ActiveRecord::Base.transaction do
+      DB.exec <<~SQL
+      INSERT INTO discourse_reactions_reactions(post_id, #{reaction_cols})
+      SELECT #{target_post.id}, #{reaction_cols}
+      FROM discourse_reactions_reactions
+      WHERE post_id = #{original_post.id}
+      SQL
+
+      reaction_id =
+        DB.query_single(
+          "SELECT id FROM discourse_reactions_reactions ORDER BY id DESC LIMIT 1",
+        ).first
```

Comment:
Sorry I'm not sure I understand this because this is just fetching the last reaction created. What if there are multiple reactions inserted from the SQL query above?